### ISSUE TITLE
Cannot instantiate Java OnlineRecognizer on Windows

### ISSUE THREAD

**Issue by ChrystianKacki** [k2-fsa/sherpa-onnx#608]
I am trying to use Sherpa-ONNX with Java API sources from:
[https://github.com/k2-fsa/sherpa-onnx/tree/master/sherpa-onnx/java-api](https://github.com/k2-fsa/sherpa-onnx/tree/master/sherpa-onnx/java-api)

I build Sherpa-ONNX JNI DLL on Windows from newest Git source with Visual Studio Build Tools 2022 with command:
```
cmake -DCMAKE_BUILD_TYPE=Release -DSHERPA_ONNX_ENABLE_BINARY=OFF -DBUILD_SHARED_LIBS=ON -DSHERPA_ONNX_ENABLE_JNI=ON -DSHERPA_ONNX_ENABLE_PORTAUDIO=OFF -DSHERPA_ONNX_ENABLE_C_API=OFF -DSHERPA_ONNX_ENABLE_WEBSOCKET=OFF ..
cmake --build . --config Release
```

I use static method `OnlineRecognizer.setSoPath(String soPath)` to load the JNI DLL.
I get message: "load so lib succeed".

I use Zipformer-transducer-based models:
csukuangfj/sherpa-onnx-streaming-zipformer-en-2023-06-21
csukuangfj/sherpa-onnx-streaming-zipformer-en-2023-06-26
sherpa-onnx-streaming-zipformer-multi-zh-hans-2023-12-12

My model config file looks like this:
```
encoder=sherpa-onnx-streaming-zipformer-en-2023-06-21/encoder-epoch-99-avg-1.onnx
decoder=sherpa-onnx-streaming-zipformer-en-2023-06-21/decoder-epoch-99-avg-1.onnx
joiner=sherpa-onnx-streaming-zipformer-en-2023-06-21/joiner-epoch-99-avg-1.onnx
tokens=sherpa-onnx-streaming-zipformer-en-2023-06-21/tokens.txt
```

In my Java app I instantiate OnlineRecognizer with config file path like this:
`OnlineRecognizer onlineRecognizer = new OnlineRecognizer(configFilePath);`

When executing above line of code, the JVM exits and I get the following error:
```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  EXCEPTION_ACCESS_VIOLATION (0xc0000005) at pc=0x00007fffa8114af0, pid=16800, tid=32480
#
# JRE version: OpenJDK Runtime Environment (21.0.2+13) (build 21.0.2+13-58)
# Java VM: OpenJDK 64-Bit Server VM (21.0.2+13-58, mixed mode, sharing, tiered, compressed oops, compressed class ptrs, g1 gc, windows-amd64)
# Problematic frame:
# V  [jvm.dll+0x444af0]
#
# No core dump will be written. Minidumps are not enabled by default on client versions of Windows
#
# An error report file with more information is saved as:
# C:\Users\chrys\IdeaProjects\asr-javafx-app\hs_err_pid16800.log
#
# If you would like to submit a bug report, please visit:
#   https://bugreport.java.com/bugreport/crash.jsp
#
```

The error report file from this log is attached: [hs_err_pid16800.log](https://github.com/k2-fsa/sherpa-onnx/files/14396466/hs_err_pid16800.log)
I see from this file, that this could be a matter of model initialization in C/C++ code, because it happens during invocation of  `OnlineRecognizer` constructor, in method: `private native long createOnlineRecognizer(Object asset, OnlineRecognizerConfig config);`

Unfortunatelly it is a low level matter and I don't know how to deal with this. I appreciate your help.

**Comment by csukuangfj** [comment]
Thank you for your detailed logs.

I will look into it tomorrow.

**Comment by ChrystianKacki** [comment]
> Thank you for your detailed logs.
> 
> I will look into it tomorrow.

Great, thank you very much.

**Comment by csukuangfj** [comment]
@ChrystianKacki 

It is fixed in #610. Please use the latest master.
